### ISSUE TITLE
Details of English AUX category

### ISSUE THREAD

**Issue by nschneid** [UniversalDependencies/docs#206]
I've written a quick Python script to convert to Universal POS tags given a PTB tree:
https://gist.github.com/nschneid/beed0bcda5b42e530011

I'm wondering if I have interpreted the [`AUX` category](http://universaldependencies.github.io/docs/en/pos/AUX_.html) correctly. E.g., there are borderline syntactic uses of "have" and "get" in "you 'll have to fight to get it resolved" (see comment with tree in the script). Is there a simple rule for identifying the syntactic environment that is treated as `AUX`?

Tagging the contributors to the English AUX definition in the docs: @spyysalo @manning Thanks!

**Comment by nschneid** [comment]
I have found https://github.com/stanfordnlp/CoreNLP/blob/master/data/edu/stanford/nlp/upos/ENUniversalPOS.tsurgeon from the Stanford parser but am unfamiliar with the TSurgeon language. @sebschu, any advice?

**Comment by sebschu** [comment]
Hi Nathan,

Yes, this is the script that we are using. We based these patterns on the UD converter patterns which have been refined over years and therefore some of them are a bit complex. But I'll try to explain which structures they match. The Tregex syntax is documented [here](http://nlp.stanford.edu/nlp/javadoc/javanlp/edu/stanford/nlp/trees/tregex/TregexPattern.html)  in case something remains unclear. There are probably also cases which are not covered by these patterns but if you simply want to reproduce our conversion, then they should be sufficient.

```
@SINV < (@VP < (/^VB/=target <  /^(?i:am|is|are|r|be|being|'s|'re|'m|was|were|been|s|ai|m|art|ar|wase)$/ ) $-- (@VP < VBD|VBN))
```

This matches a SINV node with a VP child which has a VB child that is an auxiliary verb and which has a VP with a past participle as a left sister. 

This is to find auxiliaries in sentences with VP fronting, e.g.

_Elected as director **was** Mr. Hollander._

```
@SINV < (@VP < (@VP < (/^VB/=target <  /^(?i:am|is|are|r|be|being|'s|'re|'m|was|were|been|s|ai|m|art|ar|wase)$/ )) $-- (@VP < VBD|VBN))
```

This is the same pattern as the previous one with the exception that it allows another level of nesting for the VP that contains the auxiliary verb. This is necessary for more complex VP fronting constructions such as 

_Also excluded will **be** investments in foreign companies._ 
(Note that will is also an auxiliary verb but this is matched by a different pattern.)

```
VP < VP < (/^VB.*$/=target <... {/.*/})
```

This is the main pattern for auxiliaries: A VP which has both another VP and an inflected verb as daughters.

_I **do** not know._

```
@SQ|SINV < (/^VB/=target $++ /^(?:VP)/ <... {/.*/})
```

This pattern is for questions or inverted declarative sentences with auxiliaries. It looks for a SQ or SINV with an inflected verb as a daughter that has another VP as a right sibling.

_**Did** you know that?_

With these four rules you should be able to get the same tags as we use in the latest released UD treebank.

Coming back to your original question, neither _get_ nor _have_ in this sentence are auxiliaries and thus should be tagged as `VERB` and their complements should have an `xcomp` relation.

**Comment by nschneid** [comment]
Thanks @sebschu. I see that the rule with forms of _get_ is commented out, though the documentation [here](http://universaldependencies.github.io/docs/en/pos/AUX_.html) says passive _get_ should be counted as `AUX`. Is the documentation page out of date?

**Comment by sebschu** [comment]
Yes, we only discussed this a few months ago and haven't updated the documentation since. If I recall correctly, we tag _get_ in  get + verb constructions with VERB and in get + adjective constructions with AUX.

**Comment by nschneid** [comment]
> If I recall correctly, we tag _get_ in get + verb constructions with VERB and in get + adjective constructions with AUX.

Perhaps it should be the other way around? I looked through the Web Treebank for some get + `JJ` examples. Most can be paraphrased with _become_:
- gets busy: `(VBZ gets) (ADJP-PRD (JJ busy)))`
- getting infected
- got so incredibly offended at that statement
- getting older: `(VBG getting) (ADJP-PRD (JJR older)))`; getting a lot worse; get any better

There is also
- getting (ADJ-PRD (JJ close) (PP to (NP 100,000 miles)))

where the adjective is modifying a quantity (I think the tree is debatable, but that's another conversation).

The get + `VBN` ones look more like passive auxes to me, though the ones that can be paraphrased as _had_ are borderline:
- passive-like
  - get taken in
  - getting paid
- _get_ (or _had_ paraphrase) + NP + VBN: 'induce someone else to do something for you' (extremely productive!)
  - getting it done; get my hair done
  - got our sunroom built
  - got my dog groomed

There is yet another sense of _get_ meaning 'follow instructions correctly (or incorrectly)' that is perhaps inconsistently annotated:
- get game info (JJ correct)
- get my order (VBN confused), get my friend 's order (VBN mixed) up

**Comment by sebschu** [comment]
I checked my notes again and I think you are right. I'll confirm that again at our next meeting (which probably won't be before the end of the month) but it does make a lot more sense if it is the other way round. Sorry for the confusion and I'll let you know after we discussed this.

**Comment by nschneid** [comment]
Thanks. To be clear, my script currently applies `AUX` if the candidate auxiliary has a subsequent VP sister. This even works with VP fronting because (at least in new-style PTB trees) there is a second VP with a trace. But it does not match _get_ in _get it done_.

**Comment by sebschu** [comment]
We strip all traces before processing trees (because the tagger and parser also don't generate traces and we don't want to maintain two converters) but your solution will probably still work.

And _get it done_ is in my opinion not an auxiliary but a `VERB` with a `ccomp` relation to _done_.

**Comment by manning** [comment]
Late to the party, but all Sebastian's answer are 100% ! There are a good range of semi-auxiliary constructions in English – and many other languages – but our position at the moment is to only use AUX for the core auxiliary constructions, not semi-grammaticalized things. We can re-evaluate anything, but as @sebschu suggested, our position was to make passive auxiliary _get_ an AUX but not any other uses of it – i.e., only the ones @nschneid describes as "passive-like". The argument that many uses of _get_ are like _become_ is actually an argument for treating it not as an AUX but an xcomp taking verb, since that is what we do for _become_. There is some history here. At one point in SD, we did try to recognize some other verbs, of which _become_ is the canonical case as copular verbs, so that indeed _become/seem/appear/... happy_ would be like _be happy_. But (i) this confused the hell out of users, and (ii) it was difficult in many cases to distinguish true copular uses from others, and so we moved to a narrower definition of auxiliaries. In English, it's basically just forms of _be_ and _have_ and _do_ when used as an auxiliary. But we decided to add passive uses of _get_ as basically synonymous with _be_.
